scripts/ingest-tarball.sh

```diff
@@ -73,7 +77,36 @@ then
     error "the contents type in the filename (${contents_type_dir}) does not match the contents type in the tarball (${tar_contents_type_dir})."
 fi
 
-# Ingest the tarball to the repository, use "versions" as base dir for the ingestion
+# If this is a compat layer tarball, we need to check if it's an update, and if so, use a special procedure
+if [ "${tar_contents_type_dir}" = "compat" ]
+then
+    tar_first_file="$(tar tf "${tar_file}" | head -n 1)"
+    # Get OS and architecture from path, which should look like: <version>/compat/<os>/<arch>
+    compat_os=$(echo "${tar_first_file}" | cut -d / -f 3)
+    compat_arch=$(echo "${tar_first_file}" | cut -d / -f 4)
+    # Assume that we already had a compat layer in place if there is a startprefix script in the corresponding CVMFS directory
+    if [ -f "/cvmfs/${repo}/${basedir}/${version}/compat/${compat_os}/${compat_arch}/startprefix" ];
```

Comment:
Why not also have an `else` part, the inline `exit 0` is a bit nasty perhaps?

You could set `generated_nested_catalogs` to `no` (default `yes`) in this case to skip the part below?

Reply:
Improved this (and a lot more) by refactoring the script, moved lots of functionalities to separate functions.

Reply:
This should also make it easier to have more specific checks / procedures for different tarballs, as each of them have their own function now.